tests/integration/security/ca_custom_root/trust_domain_alias_client_test.go

```diff
@@ -0,0 +1,182 @@
+package cacustomroot
+
+import (
+	"fmt"
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/tests/integration/security/util/cert"
+	"istio.io/istio/tests/integration/security/util/connection"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+const (
+	HTTPS  = "https"
+	POLICY = `
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
+metadata:
+  name: "mtls"
+spec:
+  mtls:
+    mode: STRICT
+---
+apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "server-naked"
+spec:
+  host: "*.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+`
+)
+
+// TestTrustDomainAliasClient scope:
+// The client side mTLS connection should validate the trust domain alias during secure naming validation.
+// - When server side has trust domain which defined as trust domain alias, the mTLS connection should expect success
+// - When server side has trust domain which not defined as trust domain alias, the mTLS connection should expect fail
+func TestTrustDomainAliasClient(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.trust-domain-validation-alias-client").
+		Run(func(ctx framework.TestContext) {
+			testNS := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "trust-domain-alias-client-validation",
+				Inject: true,
+			})
+
+			// Create testing certs using runtime namespace
+			//generateCerts(ctx, testNS.Name())
+			generateCerts(ctx, testNS.Name(), "foo", "server-naked-foo")
+			generateCerts(ctx, testNS.Name(), "bar", "server-naked-bar")
+			defer cleanCerts(ctx)
+
+			// Deploy 3 workloads:
+			// client: echo app with istio-proxy sidecar injected, holds default trust domain cluster.local
+			// serverNakedFoo: echo app without istio-proxy sidecar, holds custom trust domain trust-domain-foo
+			// serverNakedBar: echo app without istio-proxy sidecar, holds custom trust domain trust-domain-bar
+			var client, serverNakedFoo, serverNakedBar echo.Instance
+			echoboot.NewBuilder(ctx).
+				With(&client, echo.Config{
+					Namespace: testNS,
+					Service:   "client",
+				}).
+				With(&serverNakedFoo, echo.Config{
+					Namespace: testNS,
+					Service:   "server-naked-foo",
+					Subsets: []echo.SubsetConfig{
+						{
+							Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false),
+						},
+					},
+					ServiceAccount: true,
+					Ports: []echo.Port{
+						{
+							Name:         HTTPS,
+							Protocol:     protocol.HTTPS,
```

Comment:
Do you have to define your own port? Are there any default Echo config with 443 port we can use?

Reply:
tried without and use the auto generated port names but getting port can't find error.